File: src/main/java/io/github/oleksivio/telegram/bot/api/model/annotation/validator/FilterValidator.java
package io.github.oleksivio.telegram.bot.api.model.annotation.validator;

import io.github.oleksivio.telegram.bot.core.model.ITelegram;

public abstract class FilterValidator<T extends ITelegram>  {
    private final Class<T> targetClass;
    private final CustomValidator<T> validator;

    protected FilterValidator(Class<T> targetClass, CustomValidator<T> validator) {
        this.targetClass = targetClass;
        this.validator = validator;
    }

    public final Class<T> getFilterTarget() {
        return targetClass;
    }

    public final CustomValidator<T> getValidator() {
        return validator;
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/annotation/validator/FilterValidator.kt
package io.github.oleksivio.telegram.bot.api.model.annotation.validator

import io.github.oleksivio.telegram.bot.core.controller.handler.check.Validator
import io.github.oleksivio.telegram.bot.core.model.ITelegram
import kotlin.reflect.KClass

abstract class FilterValidator<T : ITelegram>
protected constructor(val filterTarget: KClass<T>, val validator: Validator<T>)
